partiql-ast/src/ast.rs

```diff
@@ -246,43 +185,40 @@ pub struct Expr {
     pub kind: ExprKind,
 }
 
-/// Represents an AST Node of type T with BytePosition Location
-pub type AstBytePos<T> = AstNode<T, BytePosition>;
-
-pub type BagAst = AstBytePos<Bag>;
-pub type BetweenAst = AstBytePos<Between>;
-pub type BinOpAst = AstBytePos<BinOp>;
-pub type CallAggAst = AstBytePos<CallAgg>;
-pub type CallArgAst = AstBytePos<CallArg>;
-pub type CallAst = AstBytePos<Call>;
-pub type CaseAst = AstBytePos<Case>;
-pub type FromClauseAst = AstBytePos<FromClause>;
-pub type FromLetAst = AstBytePos<FromLet>;
-pub type GroupByExprAst = AstBytePos<GroupByExpr>;
-pub type GroupKeyAst = AstBytePos<GroupKey>;
-pub type InAst = AstBytePos<In>;
-pub type JoinAst = AstBytePos<Join>;
-pub type JoinSpecAst = AstBytePos<JoinSpec>;
-pub type LetAst = AstBytePos<Let>;
-pub type LikeAst = AstBytePos<Like>;
-pub type ListAst = AstBytePos<List>;
-pub type LitAst = AstBytePos<Lit>;
-pub type OrderByExprAst = AstBytePos<OrderByExpr>;
-pub type ParamAst = AstBytePos<Param>;
-pub type PathAst = AstBytePos<Path>;
-pub type ProjectItemAst = AstBytePos<ProjectItem>;
-pub type ProjectionAst = AstBytePos<Projection>;
-pub type QueryAst = AstBytePos<Query>;
-pub type QuerySetAst = AstBytePos<QuerySet>;
-pub type SearchedCaseAst = AstBytePos<SearchedCase>;
-pub type SelectAst = AstBytePos<Select>;
-pub type SetExprAst = AstBytePos<SetExpr>;
-pub type SexpAst = AstBytePos<Sexp>;
-pub type SimpleCaseAst = AstBytePos<SimpleCase>;
-pub type SortSpecAst = AstBytePos<SortSpec>;
-pub type StructAst = AstBytePos<Struct>;
-pub type UniOpAst = AstBytePos<UniOp>;
-pub type VarRefAst = AstBytePos<VarRef>;
+pub type BagAst = AstNode<Bag>;
+pub type BetweenAst = AstNode<Between>;
+pub type BinOpAst = AstNode<BinOp>;
+pub type CallAggAst = AstNode<CallAgg>;
+pub type CallArgAst = AstNode<CallArg>;
+pub type CallAst = AstNode<Call>;
+pub type CaseAst = AstNode<Case>;
+pub type FromClauseAst = AstNode<FromClause>;
+pub type FromLetAst = AstNode<FromLet>;
+pub type GroupByExprAst = AstNode<GroupByExpr>;
+pub type GroupKeyAst = AstNode<GroupKey>;
+pub type InAst = AstNode<In>;
+pub type JoinAst = AstNode<Join>;
+pub type JoinSpecAst = AstNode<JoinSpec>;
+pub type LetAst = AstNode<Let>;
+pub type LikeAst = AstNode<Like>;
+pub type ListAst = AstNode<List>;
+pub type LitAst = AstNode<Lit>;
+pub type OrderByExprAst = AstNode<OrderByExpr>;
+pub type ParamAst = AstNode<Param>;
+pub type PathAst = AstNode<Path>;
+pub type ProjectItemAst = AstNode<ProjectItem>;
+pub type ProjectionAst = AstNode<Projection>;
+pub type QueryAst = AstNode<Query>;
+pub type QuerySetAst = AstNode<QuerySet>;
+pub type SearchedCaseAst = AstNode<SearchedCase>;
+pub type SelectAst = AstNode<Select>;
+pub type SetExprAst = AstNode<SetExpr>;
+pub type SexpAst = AstNode<Sexp>;
+pub type SimpleCaseAst = AstNode<SimpleCase>;
+pub type SortSpecAst = AstNode<SortSpec>;
+pub type StructAst = AstNode<Struct>;
+pub type UniOpAst = AstNode<UniOp>;
+pub type VarRefAst = AstNode<VarRef>;
```

Comment:
I'm curious how did you determine which parts of the ast derive `Eq`? Among this list (L188-221), only `VarRef` derives `Eq`.

Reply:
It's an oversight (likely due to some automated rewriting) that is fixed in https://github.com/partiql/partiql-lang-rust/pull/195